odoo/addons/base/models/ir_http.py

```diff
@@ -245,24 +245,36 @@ def _postprocess_args(cls, arguments, rule):
                     return cls._handle_exception(werkzeug.exceptions.NotFound())
 
     @classmethod
-    def routing_map(cls):
+    def _generate_routing_rules(cls, modules, converters):
+        return http._generate_routing_rules(modules, False, converters)
+
+    @classmethod
+    def routing_map(cls, key=None):
         if not hasattr(cls, '_routing_map'):
-            _logger.info("Generating routing map")
-            installed = request.registry._init_modules - {'web'}
+            cls._routing_map = {}
+        if key not in cls._routing_map:
+            _logger.info("Generating routing map for key %s" % str(key))
+            installed = request.registry._init_modules | set(odoo.conf.server_wide_modules)
             if tools.config['test_enable'] and odoo.modules.module.current_test:
                 installed.add(odoo.modules.module.current_test)
-            mods = [''] + odoo.conf.server_wide_modules + sorted(installed)
+            mods = sorted(installed)
             # Note : when routing map is generated, we put it on the class `cls`
             # to make it available for all instance. Since `env` create an new instance
             # of the model, each instance will regenared its own routing map and thus
             # regenerate its EndPoint. The routing map should be static.
-            cls._routing_map = http.routing_map(mods, False, converters=cls._get_converters())
-        return cls._routing_map
+            routing_map = werkzeug.routing.Map(strict_slashes=False, converters=cls._get_converters())
+            for u, e, r in cls._generate_routing_rules(mods, converters=cls._get_converters()):
+                xtra_keys = 'defaults subdomain build_only strict_slashes redirect_to alias host'.split()
```

Comment:
Missing `methods` in accepted keys

Reply:
added in routing_map.add as previously

well spotted :eye: